### PR TITLE
TF-3332 Prevent refresh when switching mailbox

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #3332 

### Demo condition
- https://github.com/linagora/tmail-flutter/pull/3357

### Web

https://github.com/user-attachments/assets/efcfd350-c05e-4490-a2eb-4cfe57e38389

### Mobile

https://github.com/user-attachments/assets/f301af91-a056-40a3-aee5-22089affb5e1

### Web

https://github.com/user-attachments/assets/86fefeed-fc97-40cd-97d8-23fb60a1c36c

### Mobile

https://github.com/user-attachments/assets/5c75360a-315f-45a1-8f92-83f308960dbf

### CODE REVIEW

**Review by dab246 — COMMENTED** [review]
- [x] Is there a way to write a unit test for it to ensure `Email/change` is not called when switching mailboxes?

**Comment by tddang-linagora** [comment]
>Is there a way to write a unit test for it to ensure Email/change is not called when switching mailboxes?

Tests added.

File: test/features/thread/presentation/controller/thread_controller_test.dart [hunk 1]
```diff
@@ -418,5 +419,55 @@ void main() {
...
+        mockMailboxDashBoardController.selectedMailbox.value = mailboxAfter;
+        await untilCalled(mockGetEmailsInMailboxInteractor.execute(
+          any,
+          any,
+          limit: anyNamed('limit'),
+          sort: anyNamed('sort'),
+          emailFilter: anyNamed('emailFilter'),
+          propertiesCreated: anyNamed('propertiesCreated'),
+          propertiesUpdated: anyNamed('propertiesUpdated'),
+          getLatestChanges: anyNamed('getLatestChanges'),
```
Comment (dab246) [inline test/features/thread/presentation/controller/thread_controller_test.dart:456]: Should be `getLatestChanges: false,`

Comment (tddang-linagora) [reply test/features/thread/presentation/controller/thread_controller_test.dart]: Done

**Review by dab246 — APPROVED** [review]
lgtm

**Review by hoangdat — APPROVED** [review]

**Pull request merged**